backends/candle/src/models/bloom.rs

```diff
@@ -0,0 +1,245 @@
+use crate::layers::LayerNorm;
+use candle::{DType, Device, Result, Tensor};
+use candle_nn::{Dropout, Embedding, Linear, VarBuilder};
+use serde::Deserialize;
+use text_embeddings_backend_core::Embedding;
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct Config {
+    pub vocab_size: usize,
+    pub hidden_size: usize,
+    pub n_layer: usize,
+    pub n_head: usize,
+    pub layer_norm_epsilon: f64,
+    pub initializer_range: f64,
+    pub use_cache: bool,
+    pub bos_token_id: usize,
+    pub eos_token_id: usize,
+    pub apply_residual_connection_post_layernorm: bool,
+    pub hidden_dropout: f64,
+    pub attention_dropout: f64,
+    pub pretraining_tp: usize,
+    pub slow_but_exact: bool,
+    pub architectures: Vec<String>,
+    pub attention_softmax_in_fp32: bool,
+    pub seq_length: usize,
+    pub pad_token_id: usize,
+    pub masked_softmax_fusion: bool,
+    pub model_type: Option<String>,
+    pub n_embed: usize,
+    pub num_attention_heads: usize,
+}
+
+fn gelu_forward(x: &Tensor) -> Result<Tensor> {
```

Comment:
A bit of a random comment but I think this is available directly in candle via `x.gelu()`.

Reply:
From the transformers docstring:
> Custom bias GELU function. Adapted from Megatron-DeepSpeed code.

The only code I could find in candle for the gelu function doesn't seem to be the same, but I'm not great at math so couldn't tell at a quick glance

Reply:
I think the implementation is the same, see [here](https://github.com/huggingface/candle/blob/9ab3f9729fc1444687578a9dc913760b0d8d9963/candle-core/src/op.rs#L599), it's just a matter of calling `x.gelu` and would probably be a lot faster than doing the full thing as it would fuse all the ops together.